middlewares/entityValidation.js

```diff
@@ -2,20 +2,31 @@ const { DOCUMENT_NOT_FOUND } = require('~/consts/errors')
 const { createError } = require('~/utils/errorsHelper')
 
 const isEntityValid = (entities) => {
+  const checkModel = async (req, models, model, idName, param) => {
+    const id = param ? req.params[idName] : req.body[idName]
+
+    if (!id) return
+
+    const document = await model.findById(id)
+
+    pushModel(document, models, model)
+  }
+
+  const pushModel = (document, models, model) => {
+    if (!document && !models.includes(model.modelName)) {
+      models.push(model.modelName)
+    }
+  }

```

Comment:
you can just use this code and not create a function:
    if (!document && !models.includes(model.modelName)) {
      models.push(model.modelName)
    }